core/api/src/domain/quiz/index.ts

```diff
@@ -0,0 +1,78 @@
+export * from "./config"
+
+import { QuizzesValue, milliSecondsBetweenSections } from "./config"
+import { QuizQuestionId } from "./index.types"
+import { QuizzesSectionsConfig } from "./sections"
+
+export interface QuizCompleted {
+  quizId: QuizQuestionId
+  createdAt: Date
+}
+
+export const QuizzesSections = QuizzesSectionsConfig.map((value, index) => ({
+  order: index,
+  ...value,
+}))
+
+type FillQuizInformationResult = {
+  currentSection: number
+  quizzes: {
+    id: QuizQuestionId
+    amount: Satoshis
+    completed: boolean
+    section: number
+    notBefore: Date | undefined
+  }[]
+}
+
+const lastSectionCompleted = (quizzesCompleted: QuizCompleted[]): number => {
+  const quizzesCompletedIds = quizzesCompleted.map((quiz) => quiz.quizId)
+  const lastQuizCompleted = quizzesCompletedIds[quizzesCompletedIds.length - 1]
+
+  // which section are we?
+  const index = QuizzesSections.findIndex((section) =>
+    section.quiz.includes(lastQuizCompleted),
+  )
+
+  // if we completed the last quiz of the current section, we move to the next
+  const section = QuizzesSections[index]?.quiz
+  if (section && section[section.length - 1] === lastQuizCompleted) return index + 1
+
+  return index === -1 ? 0 : index
+}
+
+export const fillQuizInformation = (
```

Comment:
cant this be replaced by flatMap ?

Reply:
I'm not sure, maybe it could be, but what is the issue with the current code?